Fix in-app review notification and track viewing in-app notifications

jaclync opened the pull request:
<!-- Remember about a good descriptive title. -->

Parts of #5330 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR fixes an issue where an in-app review notification automatically navigates to show review details, and tracks the event `view_in_app_push_notification_pressed` when viewing an in-app notification. The issue was caused by returning the incorrect subject to the observable 🤦🏻‍♀️

There is an [open discussion](https://github.com/woocommerce/woocommerce-ios/issues/5330#issuecomment-1014055140) on whether to replace the in-app notifications with the default iOS push notifications. Since the iOS foreground push notifications would be non-trivial work with lots of testing, I will make the changes in a later sprint if we decide to remove the current in-app notifications. For now, let's track the event to understand how many merchants find it useful.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Please test this on a device for remote push notifications to work.

- Launch the app and keep the app in the foreground
- In web, leave a review on a product in any connected store --> the app should show an in-app notice at the bottom for the new review with a "View" CTA
- Don't take any actions on the notice --> the notice should disappear and the app does not navigate to review details
- In web, leave another review on a product in any connected store --> the app should show an in-app notice at the bottom for the new review with a "View" CTA
- Tap on the "View" CTA --> in the console, there should be an event `🔵 Tracked view_in_app_push_notification_pressed, properties: [AnyHashable("push_notification_type"): "comment", ...]`
- In web, place an order in any connected store --> the app should show an in-app notice at the bottom for the new order with a "View" CTA
- Tap on the "View" CTA --> in the console, there should be an event `🔵 Tracked view_in_app_push_notification_pressed, properties: [AnyHashable("push_notification_type"): "store_order", ...]`

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

jaclync committed:
Fix in-app notification is handled upon display.

jaclync committed:
Track event when viewing an in-app notification.

jaclync committed:
Reuse the same push notification type key.

itsmeichigo reviewed (approved):
LGTM :shipit: 

There's a tracking event that I noticed in the console: `push_notification_alert_pressed` - this one is tracked when a notification comes when the app is in the foreground (tracked [here](https://github.com/woocommerce/woocommerce-ios/blob/7b6c7c49d28a11a8216098747f2fc53352eadd59/WooCommerce/Classes/Notifications/PushNotificationsManager.swift#L526)). This naming is a bit misleading and I thought it was duplicated with our new event `view_in_app_push_notification_pressed`. Do you think we can rename it? Given that this event has been tracked for a while now, I'm not sure if renaming it can affect the data tracked in the past.

jaclync committed:
Only track `pushNotificationAlertPressed` when the app is in inactive state. Also track the app state for `pushNotificationReceived`.

jaclync commented:
> There's a tracking event that I noticed in the console: push_notification_alert_pressed - this one is tracked when a notification comes when the app is in the foreground (tracked here). This naming is a bit misleading and I thought it was duplicated with our new event view_in_app_push_notification_pressed. Do you think we can rename it? Given that this event has been tracked for a while now, I'm not sure if renaming it can affect the data tracked in the past.

Great catch! it looks like a pre-existing issue where `push_notification_alert_pressed` is tracked not only when the app comes from an inactive state (when tapping on a push notification) but also when the app is active (where we show an in-app notice). In https://github.com/woocommerce/woocommerce-ios/pull/5909/commits/aa13c34d3a9e470591f748084bf217b2135c4973, I updated the tracking so that `push_notification_received` is tracked when the app receives a notification in the foreground instead of `push_notification_alert_pressed`. I also added an `app_state` property since the app could receive a push notification and fire `push_notification_received` in both "active" and "background" states. From testing, the console now shows the following when an in-app notice is shown:

```
🔵 Tracked push_notification_received, properties: [AnyHashable("push_notification_type"): "comment", AnyHashable("app_state"): 0, ...]
```

And when tapping on a push notification, `push_notification_alert_pressed` is tracked.

Feel free to take another look, I'll merge it tomorrow!

itsmeichigo commented:
The new update looks great, thank you! 💯
